nemo/collections/asr/parts/features.py

```diff
@@ -146,6 +151,50 @@ def forward(self, input_data):
         return super(STFTPatch, self).transform(input_data)[0]
 
 
+# Create helper class for STFT that yields num_frames = num_samples / hop_length
+class STFTExactPad(STFT):
+    """adapted from Prem Seetharaman's https://github.com/pseeth/pytorch-stft"""
+
+    def __init__(self, *params, **kw_params):
+        super(STFTExactPad, self).__init__(*params, **kw_params)
+        self.pad_amount = (self.filter_length - self.hop_length) // 2
+
+    def inverse(self, magnitude, phase):
```

Comment:
When do you need to use this inverse?

Reply:
I think it's used in the denoiser. Can we remove `stft_exact_pad` parameters from the preprocessors then?

Reply:
Inverse is needed for denoiser. `stft_exact_pad` is needed to know when to call `STFTExactPad`, which uses slightly different padding than `STFTPatch`